test_package/CMakeLists.txt

```diff
@@ -16,15 +16,24 @@ find_package(protobuf CONFIG REQUIRED)
 message(STATUS "Using protobuf ${protobuf_VERSION}")
 
 set(_PROTOBUF_LIBPROTOBUF protobuf::libprotobuf)
-# set(_PROTOBUF_PROTOC $<TARGET_FILE:protobuf::protoc>)
-find_program(_PROTOBUF_PROTOC protoc ${CONAN_BIN_DIRS_PROTOBUF} NO_DEFAULT_PATH)
+
+# Find protoc
+find_program(_PROTOBUF_PROTOC protoc REQUIRED)
```

Comment:
This change means that a pre-installed protoc can be found which is not desired

Reply:
For cross complication to work `find_program` needs to rely on finding `protoc` through the PATH variable, not CONAN_BIN_DIRS_PROTOBUF.
You shouldn't find the system `protoc` because the protobuf package sets up the path correctly:
https://github.com/conan-io/conan-center-index/blob/4116c939117a47bf4c4285aeec2a857e6b9b7165/recipes/protobuf/all/conanfile.py#L203

I have not been able to find any other way to do this with Conan...

The protobuf PR in CCI (https://github.com/conan-io/conan-center-index/pull/4556) makes sure
the path is setup correctly when using dual profiles (to make sure you find the build protoc).